docs/CONTRIBUTE_CODE.md

```diff
@@ -0,0 +1,140 @@
+* <span style="color:blue;background-color:#d4d4f7">**New**</span> - implemented a new feature.
+* <span style="color:blue;background-color:#d4d4f7">**Docs**</span> - changes to documentation only.
+* <span style="color:blue;background-color:#d4d4f7">**Build**</span> - changes to build process only, updating the dependency libs etc.
+* <span style="color:blue;background-color:#d4d4f7">**Chore**</span> - for refactoring, adding tests, paying tech debt etc. (anything that isn’t user-facing).
+
+
+Use the labels of the issue you are working on to determine the best tag.
+
+The message summary should be a one-sentence description of the change, and it must be 72 characters in length or shorter. Please make the short description concise. If the pull request addresses an issue, then the issue number should be mentioned at the end. If the commit doesn’t completely fix the issue, then use (refs #1234) instead of (fixes #1234).
+
+**Note**: please squash you changes in one commit before firing the pull request. One commit in one PR keeps the git history clean.
+
+#### Step 3: Rebase onto upstream
+
+Before you send the pull request, be sure to rebase onto the upstream source. This ensures your code is running on the latest available code. We prefer rebase instead of merge when upstream changes. Rebase keeps the git history clearer.
+```bash
+git fetch upstream
+git rebase upstream/master
+```
+
+#### Step 4: Run the tests
+
+The is a place holder as well. We should write about 
+* how to run unit test
+* how to run funcional test
+* what is the acceptance criteria about the test
+* how to add test cases
+
+
+#### Step 5: Push your change
+Before pushing the changes, double check your changes follows the [code style](#code-style-and-formatting-tools) and all tests are passed.
+
+
+Next, push your changes to your clone:
+```bash
+git push origin fix/issue123
+```
+
+#### Step 6: Submit the pull request
+
+Before creating a pull request, here are some recommended **check points**. 
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a
+   build.
+2. Update the splunk_eventgen/README/eventgen.conf.spec with details of changes to the interface, this includes new environment
+   variables, exposed ports, useful file locations and container parameters. Also, update the necessary documentation.
+3. Make sure the build is successful and all test cases are passed.
+4. You may merge the Pull Request in once you have the sign-off of two other developers, or if you
+   do not have permission to do that, you may request the second reviewer to merge it for you.
+
+
+Next, create a pull request from your branch to the eventgen develop branch.
+Mark @lephino , @arctan5x , @jmeixensperger , @li-wu , @GordonWang as the reviewers.
+
+
+
+## Code style and formatting tools
+
+Eventgen is written with python. And the python code style is adoptted.
```

Comment:
"Since Eventgen is written in python, we apply a coding style based on PEP8 (link to pep8)."

Reply:
Gotcha!